domain/storageprovisioning/service/filesystem.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/machine"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain/life"
+	"github.com/juju/juju/domain/storageprovisioning"
+	"github.com/juju/juju/internal/errors"
+)
+
+// FilesystemState defines the interface required for performing filesystem
+// provisioning operations in the model.
+type FilesystemState interface {
+	// GetFilesystemAttachmentIDs returns the
+	// [storageprovisioning.FilesystemAttachmentID] information for each
+	// filesystem attachment uuid supplied. If a uuid does not exist or isn't
+	// attached to either a machine or a unit then it will not exist in the
+	// result.
+	GetFilesystemAttachmentIDs(ctx context.Context, uuids []string) (map[string]storageprovisioning.FilesystemAttachmentID, error)
```

Comment:
Can we add a brief sentence to mention why we don't care about missing uuids. I thought it weird until I read the doc on the service method which calls this which explains it. Ideally we'd not need to do this but it is what it is.

Reply:
Yep good point.